mutating-admission-controller/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/golang/glog"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+const (
+	port = "8080"
+)
+
+var (
+	tlscert, tlskey string
+)
+
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
+type admitFunc func(v1.AdmissionReview) *v1.AdmissionResponse
+
+func toAdmissionResponse(err error) *v1.AdmissionResponse {
+	return &v1.AdmissionResponse{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
+
+func serveMutateServiceAccount(w http.ResponseWriter, r *http.Request) {
+	serve(w, r, mutateServiceAccount)
+}
+
+func serve(w http.ResponseWriter, r *http.Request, admit admitFunc) {
+	var body []byte
+	if r.Body != nil {
```

Comment:
I think if body is nil, we could pass it directly to JSON decoder and skip all the checks. Not sure if we care about specific error messages being returned.